bundles/nl.esi.comma.testspecification/src/nl/esi/comma/testspecification/abstspec/generator/RefKVPGenerator.xtend

```diff
@@ -51,48 +51,56 @@ import nl.esi.comma.testspecification.testspecification.RunStep
 
 class RefKVPGenerator {
     def generateRefKVP(AbstractTestDefinition atd) {
-        var txt = 
-        '''
-        matlab_calls=[
-            «FOR testseq : atd.testSeq  SEPARATOR ',' »
-                «FOR step : testseq.step.filter(AssertionStep) SEPARATOR ',' »
-                «FOR mlcal : step.asserts.flatMap(a| a.ce).flatMap(i| i.constr).filter(GenericScriptBlock) SEPARATOR ',' »
-                {
-                    "id":"«getScriptId(mlcal,step)»", 
-                    "script_path":"«mlcal.params.scriptApi»",
-                    "parameters":[
-                        {
-                            "type": "OUTPUT",
-                            "value":"«mlcal.params.scriptOut»"
-                        }«FOR param : mlcal.params.scriptArgs BEFORE "," SEPARATOR ","»
-                        {
-                            «IF param instanceof ScriptParameterNamed»"name": "«getScriptParamName(param)»",«ENDIF»
-                            "type": "«getScriptParamType(param)»"«IF param instanceof ScriptParameterWithValue»,
-                            "value": «expression(param, step)»«ENDIF»
-                        }«ENDFOR»
-                    ]
-                }
-                «ENDFOR»
+        var constraints = atd.testSeq.flatMap(i|i.step).filter(AssertionStep)
```

Comment:
Lambda expressions are encapsulated within square brackets in Xtext, instead of round brackets. In addition you can also omit the `i` variable and use the implicit `it` variable.

```suggestion
        var constraints = atd.testSeq.flatMap[step].filter(AssertionStep)
```

Reply:
Also applies for other `flatMap` calls